pkg/uquery/mappings/mappings.go

```diff
@@ -44,27 +44,40 @@ func Request(analyzers map[string]*analysis.Analyzer, data map[string]interface{
 
 	mappings := meta.NewMappings()
 	for field, prop := range properties {
+		var propFields map[string]meta.Property
+
 		prop, ok := prop.(map[string]interface{})
 		if !ok {
 			return nil, errors.New(errors.ErrorTypeParsingException, fmt.Sprintf("[mappings] properties [%s] should be an object", field))
 		}
+
 		if v, ok := prop["properties"]; ok {
 			if _, ok := v.(map[string]interface{}); !ok {
 				return nil, errors.New(errors.ErrorTypeParsingException, fmt.Sprintf("[mappings] properties [%s] should be an object", field))
 			}
+
 			if subMappings, err := Request(analyzers, prop); err == nil {
 				for k, v := range subMappings.ListProperty() {
 					mappings.SetProperty(field+"."+k, v)
 				}
 			} else {
 				return nil, err
 			}
+
 			continue
 		}
+
+		if v, ok := prop["fields"]; ok {
+			if propFields, ok = v.(map[string]meta.Property); !ok {
+				return nil, errors.New(errors.ErrorTypeParsingException, fmt.Sprintf("[mappings] property.fields [%s] should be an object", field))
+			}
+		}
```

Comment:
I'm not sure if this can work, Can you add some tests for this?

Reply:
Sure, can you explain why you think this may not work?